R/data_generation.R

```diff
@@ -0,0 +1,73 @@
+
+#' Generate time-to-event data
+#'
+#' Function generates time-to-event data. We follow the data-generating mechanism from 
+#' Bender et al. and Remiro-Azocar et al. 2021 to simulate Weibull-distribued survival times 
+#' under a proportional hazards parameterization. As per Remiro-Azocar et al, we set the 
+#' default scale parameter (lambda) to 8.5 and shape parameter (nu) to 1.3 to reflect frequently
+#' observed mortality trends in metastatic cancer patients. Censoring times are generated from 
+#' exponential distribution where the rate parameter is 0.96 is selected to achieve a censoring rate
+#' of 35% under the active treatment at baseline (with the values of the covariates set to zero).
+#'
+#' @param N sample size
+#' @param lambda scale parameter in baseline hazard function
+#' @param nu shape parameter in baseline hazard function
+#' @param rateC rate parameter of the exponential distribution of censoring times
+#' @param cor_X correlation between covariates
+#' @param marginal_X_mean marginal mean of covariates
+#' @param marginal_X_sd marginal standard deviation of covariates
+#' @param beta vector of coefficients for prognostic factors
+#' @param gamma vector of coefficients for effect modifiers
+#' @param d0 coefficient for the treatment effect when X = 0
+#' @param seed seed used for simulation reproducibility
+#'
+#' @return data.frame with survival time, censor (event) status, treatment indicator, and covariates
+#'
+#' @references Remiro-Azocar et al. 2021
+#'
+#' @export
+
+
+generate_survival_data <- function(N = 150, ncov = 4, lambda = 8.5, nu = 1.3, rateC = 0.96, 
+                                   cor_X = 0.35, marginal_X_mean = 0.6, marginal_X_sd = 0.2,
+                                   beta = NULL, gamma = NULL, d0 = log(0.25), seed = 1){
+  
+  set.seed(seed)
+  
+  #if beta and gamma is not specified, give it default values
+  if(is.null(beta)){
+    beta <- rep(-log(0.67), ncov)
+  }
+  
+  if(is.null(gamma)){
+    gamma <- rep(-log(0.67), ncov)
+    gamma[1:round(ncov/2)] <- 0
```

Comment:
What should happen if `ncov=1`?

Try to avoid this `1:x` construction, better to use `seq_int(x)`, so we don't get problems with `x=0`